content/posts/2024-08-31-half-precision-in-llvm-libc.md

```diff
@@ -0,0 +1,74 @@
+---
+author: "Nicolas Celik"
+date: "2024-08-31"
+tags: ["GSoC", "libc"]
+title: "GSoC 2024: Half-precision in LLVM libc"
+---
+
+C23 defines new floating-point types, such as `_Float16`, which corresponds to
+the binary16 format from IEEE Std 754, also known as "half-precision," or FP16.
+C23 also defines new variants of the C standard library's math functions
+accordingly, such as `fabsf16` to get the absolute value of a `_Float16`.
+
+The "Half-precision in LLVM libc" Google Summer of Code 2024 project aimed to
+implement these new `_Float16` math functions in LLVM libc, making it the first
+known C standard library implementation to implement these C23 functions.
+
+We split math functions into two categories: basic operations and higher math
+functions. The current implementation status of math functions in LLVM libc can
+be viewed at https://libc.llvm.org/math/index.html#implementation-status.
+
+The exact goals of this project were to:
+
+1. Setup generated headers properly so that the `_Float16` type and `_Float16`
+   functions can be used with various compilers and architectures.
+2. Add generic implementations of `_Float16` basic operations for supported
+   architectures.
+3. Add optimized implementations of `_Float16` basic operations for specific
+   architectures using special hardware instructions and compiler builtins
+   whenever possible.
+4. Add generic implementations of as many `_Float16` higher math functions as
+   possible. We knew we would not have enough time to implement all of them.
+
+## Work done
```

Comment:
It would be great if some code references were added. E.g. links to commits, PRs, etc.